akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -165,23 +164,75 @@ import com.fasterxml.jackson.dataformat.cbor.CBORFactory
     import akka.util.ccompat.JavaConverters._
     conf.getStringList("whitelist-class-prefix").asScala.toVector
   }
+  private val typeInManifest: Boolean = conf.getBoolean("type-in-manifest")
+  // Calculated eagerly so as to fail fast
+  private val configuredDeserializationType: Option[Class[_ <: AnyRef]] = conf.getString("deserialization-type") match {
+    case "" => None
+    case className =>
+      system.dynamicAccess.getClassFor[AnyRef](className) match {
+        case Success(c) => Some(c)
+        case Failure(_) =>
+          throw new IllegalArgumentException(
+            s"Cannot find deserialization-type [$className] for Jackson serializer [$bindingName]")
+      }
+  }
 
   // This must lazy otherwise it will deadlock the ActorSystem creation
   private lazy val serialization = SerializationExtension(system)
 
+  // This must be lazy since it depends on serialization above
+  private lazy val deserializationType: Option[Class[_ <: AnyRef]] = if (typeInManifest) {
```

Comment:
One thing - due to the laziness here, if you have zero or many bindings declared, you won't get the error until the first time you attempt to deserialize a message. It would be much better to fail eagerly on startup. We could do that by reading the config ourselves (by instantiating `Serialization.Settings` from config). Is it worth it to do that?

Also, I just noticed we're not looking at the programmatically configured bindings, only the bindings declared in configuration. I don't think that's a problem, since there would be no way to configure this serializer or its bindings programatically, this serializer depends on an actor system, so can't be passed to the actor system bootstrap. Hence only bindings declared in configuration would be relevant here.

Reply:
The programmatically defined serializers can depend on ActorSystem because it's a factory from ActorSystem.
See `SerializationSetup`.

This should perhaps look in `serialization.bindings` instead?

I don't think the lazy error is a big deal, let's keep it lazy.

We could also consider to always require `deserialization-type` instead of looking for a single binding, but it's convenient to use the binding in many cases.

Reply:
I've updated the PR to use `serialization.bindings`.